web/modules/custom/server_general/src/Plugin/EntityViewBuilder/NodeGroup.php

```diff
@@ -0,0 +1,237 @@
+   *   Render array
+   *
+   * @throws \IntlException
+   */
+  protected function buildHeader(NodeInterface $entity): array {
+    $elements = [];
+
+    // Build page title.
+    $elements[] = $this->buildConditionalPageTitle($entity);
+
+    // Show the node type as a label.
+    $node_type = NodeType::load($entity->bundle());
+    $elements[] = $this->buildLabelsFromText([$node_type->label()]);
+
+    // Date.
+    $timestamp = $this->getFieldOrCreatedTimestamp($entity, 'field_publish_date');
+    $element = IntlDate::formatPattern($timestamp, 'long');
+
+    // Make text bigger.
+    $elements[] = $this->wrapTextResponsiveFontSize($element, 'lg');
+
+    $elements = $this->wrapContainerVerticalSpacing($elements);
+
+    return $this->wrapContainerNarrow($elements);
+  }
+
+  /**
+   * Build the Main content and the sidebar.
+   *
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return array
+   *   Render array
+   *
+   * @throws \Drupal\Core\Entity\EntityMalformedException
+   */
+  protected function buildMainAndSidebar(NodeInterface $entity): array {
+    $main_elements = [];
+    $sidebar_elements = [];
+    $social_share_elements = [];
+
+    // Get the body text, wrap it with `prose` so it's styled.
+    // NOTE: Function buildProcessedText has wrong default field value:
+    // "field_body" instead of "body" so we pass correct value.
+    $main_elements[] = $this->buildProcessedText($entity, 'body');
+
+    // Get the tags, and social share.
+    $sidebar_elements[] = $this->buildTags($entity);
+
+    // Show the featured image.
+    $medias = $entity->get('field_featured_image')->referencedEntities();
+    $sidebar_elements[] = $this->wrapContainerVerticalSpacing($this->buildEntities($medias, 'full'));
+
+    // Add a line separator above the social share buttons.
+    $social_share_elements[] = $this->buildLineSeparator();
+    $social_share_elements[] = $this->buildSocialShare($entity);
+
+    $sidebar_elements[] = $this->wrapContainerVerticalSpacing($social_share_elements);
+
+    return [
+      '#theme' => 'server_theme_main_and_sidebar',
+      '#main' => $this->wrapContainerVerticalSpacingBig($main_elements),
+      '#sidebar' => $this->wrapContainerVerticalSpacingBig($sidebar_elements),
+    ];
+
+  }
+
+
+  /**
+   * Helper - Checks OG group subscribe access and membership status.
+   *
+   * If user is authenticated and NOT group admin and is not group member
+   * already, we enable OG group field content override.
+   *
+   * @param \Drupal\node\NodeInterface $entity
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *
+   * @return bool
+   *   Returns TRUE if OG group field should be overridden.
+   */
+  private function checkMembershipAndSubscribeAccess(NodeInterface $entity, AccountInterface $account) {
+    $override = FALSE;
+
+    if ($account->isAuthenticated() && ($entity instanceof EntityOwnerInterface) && ($entity->getOwnerId() != $account->id())) {
```

Comment:
`if (!$account->isAuthenticated()) {
   // User is not auth ...
  return FALSE;
}
```